src/ServiceControl.Audit/Recoverability/FailedMessagesFeature.cs

```diff
@@ -28,21 +30,36 @@ public void Enrich(AuditEnricherContext context)
                 var headers = context.Headers;
                 var isOldRetry = headers.TryGetValue("ServiceControl.RetryId", out _);
                 var isNewRetry = headers.TryGetValue("ServiceControl.Retry.UniqueMessageId", out var newRetryMessageId);
+                var isAckHandled = headers.ContainsKey("ServiceControl.Retry.AcknowledgementSent");
+                var hasAckQueue = headers.TryGetValue("ServiceControl.Retry.AcknowledgementQueue", out var ackQueue);
 
                 var hasBeenRetried = isOldRetry || isNewRetry;
 
                 context.Metadata.Add("IsRetried", hasBeenRetried);
 
-                if (!hasBeenRetried)
+                if (!hasBeenRetried || isAckHandled)
                 {
                     return;
                 }
 
-                context.AddForSend(new MarkMessageFailureResolvedByRetry
+                if (hasAckQueue && isNewRetry)
```

Comment:
Do we need `isNewRetry`? If the message has `ack-queue` header is had to be generated from a version of SC that is based on the `UniqueMessageId`. Right?